### ISSUE TITLE
Add Tracing, HealthChecks, Metrics

### ISSUE THREAD

**Issue by timheuer** [milvus-io/milvus-sdk-csharp#85]
**Describe the solution you'd like**
Would like the C# SDK for Milvus to have Tracing/HealthChecks/Metrics.  For .NET Aspire, the components are desired to have all of these 'by default' for consumers.  .NET Aspire is introducing hosting support for Milvus and wrapping the Milvus C# Client in a component and would like these to be exposed more. See: https://github.com/dotnet/aspire/tree/main/src/Components

**Comment by roji** [comment]
For tracing - makes absolute sense to me.

For metrics, note that Milvus emits these on the server side ([docs](https://milvus.io/docs/metrics_dashboard.md), [docs](https://milvus.io/docs/monitor_overview.md)); using this probably makes more sense than on the cilent side (since you get fully aggregated metrics for the entire server). It's probably worth checking how easy it would be to add these server metrics to Aspire...

For the health check, MilvusClient already exposes a HealthCheck() API - can that be used?

**Comment by timheuer** [comment]
RE: HealthChecks -- let me see if I can add that easily -- we've been leveraging the AspNetCore.HealthChecks project as primary generally and there isn't a Milvus one there. https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/2214

RE: Metrics, the Aspire pattern has been that the Client provides the signal still not a direct from the server.  eg: https://github.com/dotnet/aspire/blob/main/src/Components/Aspire.MySqlConnector/AspireMySqlConnectorExtensions.cs#L99

**Comment by timheuer** [comment]
Got the HealthChecks implemented for Aspire.

**Comment by roji** [comment]
> RE: Metrics, the Aspire pattern has been that the Client provides the signal still not a direct from the server

Yeah, that's true, but I think that's more because traditional database servers (PostgreSQL, MySQL, SQL Server...) simply don't emit OpenTelemetry metrics (yet)... Where server metrics do exist (like in the Milvus case, it seems), by default I'd expect them to be the better thing - the server has more meaningful/interesting/internal metrics it can report, and in general, you're interested in your server performance (so aggregated for all clients) rather than a particular client instance, I'd think...
